dashboard/app/models/follower.rb

```diff
@@ -20,9 +20,9 @@
 # Join table defining student-teacher relationships for Users
 # (student_user is the student, user is the teacher)
 class Follower < ActiveRecord::Base
-  belongs_to :user
-  belongs_to :student_user, foreign_key: "student_user_id", class_name: User
   belongs_to :section
+  has_one :user, through: :section
```

Comment:
Humorously enough, this makes it so that the recent `user.id == user_id` validation would now be "correct".